packages/block-library/src/gallery/tiles.native.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { Children } from '@wordpress/element';
+
+const containerStyle = {
+	flexDirection: 'row',
+	flexWrap: 'wrap',
+};
+
+const tileStyle = {
+	overflow: 'hidden',
+	flexDirection: 'row',
+	alignItems: 'center',
+	borderColor: 'transparent',
+};
+
+function Tiles( props ) {
+	const {
+		columns,
+		children,
+		groutSpacing = 10,
+	} = props;
+
+	const tileCount = Children.count( children );
+	const lastTile = tileCount - 1;
+	const lastRow = Math.floor( lastTile / columns );
+
+	const wrappedChildren = Children.map( children, ( child, index ) => {
+		const row = Math.floor( index / columns );
+		const rowLength = row === lastRow ? ( lastTile % columns ) + 1 : columns;
+		const indexInRow = index % columns;
+
+		return (
+			<View style={ [ tileStyle, {
+				width: ( ( 1 / rowLength ) * 100 ) + '%',
+				borderLeftWidth: groutSpacing * ( indexInRow / rowLength ),
+				borderRightWidth: groutSpacing * ( 1 - ( ( indexInRow + 1 ) / rowLength ) ),
```

Comment:
Comparing indexes and lengths always gets annoying 😁 In this case, if you defined `lastIndexInRow = rowLength - 1` and use that as the denominator in both, you can skip the index offset and the code looks clearer IMO: the right border would read more clearly as `groutSpacing - borderLeft`.

Reply:
Using `rowLength - 1` as a denominator can be problematic for rows that only have one tile (divide by zero). The underlying problem I've struggled with in simplifying this further is that the left and right spacings for a given tile are not linearly dependent: i.e. right border cannot be expressed solely as a function of the spacing and the left border. Instead, the right border is complementary with the left border of the _next_ tile (hence the fencepost calculations). I've added an explanation in the code comment.